### PR TITLE
chore(providers): rename provider binaries with -provider suffix

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit updates the binaries for providers with a suffix (`-provider`) so that it's clear they're provider binaries as we add more binaries to the project.

### CODE REVIEW

File: .github/workflows/provider.yml [hunk 1]
```diff
@@ -87,28 +87,28 @@ jobs:
 
     - name: Push `${{ inputs.name }}` provider `${{ github.sha }}` tag to GitHub Container Registry
       if: startswith(github.ref, format('refs/tags/provider-{0}-v', inputs.name)) || github.ref == 'refs/heads/main'
-      run: ./wash push ghcr.io/${{ steps.ctx.outputs.owner }}/${{ inputs.name }}:${{ github.sha }} "${{ inputs.name }}.par.gz"
+      run: ./wash push ghcr.io/${{ steps.ctx.outputs.owner }}/${{ inputs.name }}-provider:${{ github.sha }} "${{ inputs.name }}.par.gz"
```
[the remark this answers is not in the document]
Comment (brooksmtownsend) [reply .github/workflows/provider.yml:90]: Second this, I would prefer to keep the OCI references simpler to type

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -1,21 +0,0 @@
-name: nix-flake-update
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply .github/workflows/update.yml]: Please note that this is still a Draft pull request -- this was done to speed up the tests of the provider workflow and exclude all the other workflows, while it was in Draft.

File: .github/workflows/washboard.yml [hunk 3]
```diff
@@ -1,107 +0,0 @@
-name: washboard
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply .github/workflows/washboard.yml]: Please note that this is still a Draft pull request -- this was done to speed up the tests of the provider workflow and exclude all the other workflows, while it was in Draft.

File: .github/workflows/artifacthub.yml [hunk 4]
```diff
@@ -1,45 +0,0 @@
-name: ArtifactHub Publish
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply .github/workflows/artifacthub.yml]: Please note that this is still a Draft pull request -- this was done to speed up the tests of the provider workflow and exclude all the other workflows, while it was in Draft.

File: .github/workflows/wash.yml [hunk 5]
```diff
@@ -1,103 +0,0 @@
-name: wash
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply .github/workflows/wash.yml]: Please note that this is still a Draft pull request -- this was done to speed up the tests of the provider workflow and exclude all the other workflows, while it was in Draft.

File: .github/workflows/wasmcloud-chart.yml [hunk 6]
```diff
@@ -1,88 +0,0 @@
-name: wasmcloud-chart
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply .github/workflows/wasmcloud-chart.yml]: Please note that this is still a Draft pull request -- this was done to speed up the tests of the provider workflow and exclude all the other workflows, while it was in Draft.

File: .github/workflows/wasmcloud.yml [hunk 7]
```diff
@@ -144,17 +144,17 @@ jobs:
...
+      - run: move "target/release/blobstore-s3-provider.exe" "artifact/bin/blobstore-s3-provider.exe"
+      - run: move "target/release/http-client-provider.exe" "artifact/bin/http-client-provider.exe"
+      - run: move "target/release/http-server-provider.exe" "artifact/bin/http-server-provider.exe"
+      - run: move "target/release/keyvalue-redis-provider.exe" "artifact/bin/keyvalue-redis-provider.exe"
+      - run: move "target/release/keyvalue-vault-provider.exe" "artifact/bin/keyvalue-vault-provider.exe"
+      - run: move "target/release/lattice-controller-provider.exe" "artifact/bin/lattice-controller-provider.exe"
+      - run: move "target/release/messaging-kafka-provider.exe" "artifact/bin/messaging-kafka-provider.exe"
+      - run: move "target/release/messaging-nats-provider.exe" "artifact/bin/messaging-nats-provider.exe"
+      - run: move "target/release/wash-provider.exe" "artifact/bin/wash-provider.exe"
+      - run: move "target/release/wasmcloud-provider.exe" "artifact/bin/wasmcloud-provider.exe"
```
Comment (rvolosatovs) [inline .github/workflows/wasmcloud.yml:157]: this does not look right

Comment (vados-cosmonic) [reply .github/workflows/wasmcloud.yml]: Ah you're right, that's not right -- overzealous replace, added some spacing and fixed this

**Review by rvolosatovs — APPROVED** [review]

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**